confluent_kafka/avro/avro_producer.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+#
+# Copyright 2016 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+
+from confluent_kafka.avro import ClientError
+from confluent_kafka.avro.cached_schema_registry_client import CachedSchemaRegistryClient
+from confluent_kafka.avro.serializer import SerializerError
+from confluent_kafka.avro.serializer.message_serializer import MessageSerializer
+
+log = logging.getLogger(__name__)
+class AvroProducer(object):
+    '''
+        Kafka Producer client which does avro schema encoding to messages.
+        Handles schema registration, Message serialization.
+
+        Constructor takes below parameters
+
+        @:param: producer: confluent_kafka.Producer object
+        @:param: message_serializer: Message Serializer object
+    '''
+    def __init__(self, producer, schema_registry_url, key_schema = None, value_schema = None):  # real signature unknown; restored from __doc__
+        self._producer = producer
+        _cash_client= CachedSchemaRegistryClient(url=schema_registry_url)
+        self._serializer =  MessageSerializer(_cash_client)
+        self.key_schema = key_schema
+        self.value_schema = value_schema
+
+    def produce(self, **kwargs):
+        '''
+            Sends message to kafka by encoding with specified avro schema
+            @:param: topic: topic name
+            @:param: value: A dictionary object
+            @:param: value_schema : Avro schema for value
+            @:param: key: A dictionary object
+            @:param: key_schema : Avro schema for key
+            @:exception: SerializerError
+        '''
+        # get schemas from  kwargs if defined
+        key_schema = kwargs.pop('key_schema', None)
+        value_schema = kwargs.pop('value_schema', None)
+        topic= kwargs.pop('topic', None)
```

Comment:
Giving up on commenting on this, but yo uhave a bunch of differen space styles:

    style= one
    style = two
    style=three

Please just use `style = two`